jmclient/jmclient/blockchaininterface.py

```diff
@@ -447,29 +481,10 @@ def estimate_fee_per_kb(self, N):
             # if it is not able to make an estimate. We insist that
             # the 'feerate' key is found and contains a positive value:
             if estimate and estimate > 0:
-                estimate_in_sat = btc.btc_to_sat(estimate)
-                retval = random.uniform(estimate_in_sat,
-                    estimate_in_sat * float(1 + tx_fees_factor))
-                break
-        else:  # cannot get a valid estimate after `tries` tries:
-            fallback_fee = 10000
-            retval = random.uniform(fallback_fee,
-                fallback_fee * float(1 + tx_fees_factor))
-            log.warn("Could not source a fee estimate from Core, " +
-                     "falling back to default: " +
-                     btc.fee_per_kb_to_str(fallback_fee) + ".")
-
-        if retval < mempoolminfee_in_sat:
-            log.info("Using this mempool min fee as tx feerate: " +
-                btc.fee_per_kb_to_str(mempoolminfee_in_sat) + ".")
-            return int(mempoolminfee_in_sat_randomized)
-        else:
-            msg = "Using bitcoin network feerate for " + str(N) + \
-                " block confirmation target"
-            if tx_fees_factor != 0:
-                msg = msg + " (randomized for privacy)"
-            log.info(msg + ": " + btc.fee_per_kb_to_str(retval))
-            return int(retval)
+                return btc.btc_to_sat(estimate)
+            else:  # cannot get a valid estimate after `tries` tries:
+                log.warn("Could not source a fee estimate from Core")
+                return None
```

Comment:
in 68352d047a636496ec5272c41b4d168ecf481c40:

This `else` will stop the loop, which I don't think it is what we want (as per the comment `cannot get a valid estimate after tries tries`)

I believe in the old code this `else` was used like [this](https://stackoverflow.com/questions/9979970/why-does-python-use-else-after-for-and-while-loops).

Reply:
Identation issue, fixed in 3a8f456bd9d4e7297cb502224757a2537fc1ffb7.

Reply:
I'm not sure how the `else` works in this case (with `return` inside the loop instead of `break`), but in any case I think we don't need it anymore, right?

We can do just:

```python3
    def _estimate_fee_basic(self, tx_fees: int) -> Optional[int]:
        # Special bitcoin core case: sometimes the highest priority
        # cannot be estimated in that case the 2nd highest priority
        # should be used instead of falling back to hardcoded values
        tries = 2 if tx_fees == 1 else 1
        for i in range(tries):
            rpc_result = self._rpc('estimatesmartfee', [tx_fees + i])
            if not rpc_result:
                # in case of connection error:
                return None
            estimate = rpc_result.get('feerate')
            # `estimatesmartfee` will currently return in the format
            # `{'errors': ['Insufficient data or no feerate found'], 'blocks': N}`
            # if it is not able to make an estimate. We insist that
            # the 'feerate' key is found and contains a positive value:
            if estimate and estimate > 0:
                return btc.btc_to_sat(estimate)
        # cannot get a valid estimate after `tries` tries:
        log.warn("Could not source a fee estimate from Core")
```

Reply:
Yes, that `else` is actually not needed anymore, code is easier to read without it. Addressed in 36929c0453aa88dd3257162f1462c36f4ddd785b.